src/main/java/com/example/GymProject/dao/TrainerDao.java

```diff
@@ -27,61 +30,37 @@ public List<Trainer> findAll() {
     }
 
     public void create(Trainer trainer) {
-        try {
-            if (trainer.getUserId() == null) {
-                logger.warning("Trainer userId is null");
-                throw new IllegalArgumentException("UserId cannot be null");
-            }
-            trainerMap.put(trainer.getUserId(), trainer);
-            logger.info("Inserted New Trainer");
-        } catch (Exception e){
-            logger.severe("Error While Inserting New Value");
-            throw new IllegalArgumentException(e);
-        }
+        Assert.notNull(trainer.getUserId(), "Trainer userId is null");
+        trainerMap.put(trainer.getUserId(), trainer);
+        logger.info("Inserted New Trainer");
+
     }
 
     public Trainer select(String key) {
         Trainer trainer;
-        try {
-            if (trainerMap.containsKey(key)) {
-                logger.info("Trainer found");
-                trainer = trainerMap.get(key);
-            } else {
-                logger.warning("Wrong Key, Trainer not Found");
-                throw new IllegalArgumentException("Trainer Not Found");
-            }
-            return trainer;
-        } catch (Exception e) {
-            throw new IllegalArgumentException(e);
-        }
+        Assert.isTrue(trainerMap.containsKey(key), "Wrong Key, Trainer not Found");
+        logger.info("Trainer found");
+        trainer = trainerMap.get(key);
+        return trainer;
+
     }
 
     public void update(String key, Trainer trainer) {
-        try {
-            if (key == null) {
-                logger.warning("Trainer userId is null");
-                throw new IllegalArgumentException("UserId cannot be null");
-            }
-            trainerMap.put(key, trainer);
-            logger.info("Trainer Updated");
-        } catch (Exception e){
-            logger.warning("Trainer has not been Updated");
-            throw new IllegalArgumentException(e);
-        }
+        Assert.notNull(key,"Trainer userId is null");
+        Assert.notNull(trainer,"The trainer parameter cannot be null");
+        trainerMap.put(key, trainer);
+        logger.info("Trainer Updated");
+
     }
+
     public void delete(String key){
-        try {
-            if (trainerMap.containsKey(key)){
-                trainerMap.remove(key);
-                logger.info("Trainer Removed Successfully!");
-            }else {
-                logger.warning("Trainer has not been Removed");
-                throw new IllegalArgumentException("Wrong Key");
-            }
-        } catch (Exception e){
-            throw new IllegalArgumentException(e);
-        }
+        Assert.notNull(key,"UserId cannot be null");
+        Assert.isTrue(trainerMap.containsKey(key), "Wrong Key");
+        trainerMap.remove(key);
+        logger.info("Trainer Removed Successfully!");
+
     }
+
     public boolean containsKey(String key) {
```

Comment:
I guess, this should be `protected` instead of `public` to indicate that this method should be used by tests, placed in the same package, right?